firebase.agent.nut

```diff
@@ -594,16 +610,17 @@ class Firebase {
     // process the http response accordingly
     function _processResponse (request,callback) {
         request.sendasync(function(res) {
-            local data = res.body ;
+            local data = res.body;
             try {
-                data = http.jsondecode(data);
                 if (200 <= res.statuscode && res.statuscode < 300) {
-                    callback(null,data);
+                    data = http.jsondecode(data);
```

Comment:
Do we ever get a response from firebase where the status code is in the 200's but the response.body is an empty string? Do we want this situation to bubble up as a json parsing error?